ts/features/cieLogin/components/CieIdLoginWebView.tsx

```diff
@@ -113,14 +114,22 @@ const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
   const checkIfUrlIsWhitelisted = useCallback(
     (url: string) => {
       // Checks if the URL starts with one of the valid URLs
-      const isUrlValid = WHITELISTED_URLS.some(baseUrl =>
-        url.startsWith(baseUrl)
-      );
-      // Checks whether the URL starts with the specified string
-      if (isUrlValid) {
-        // Set the URL as valid
-        setAuthenticatedUrl(url);
-      } else {
+
+      try {
+        const { hostname, protocol } = new URL(url);
+
+        const isDomainValid = WHITELISTED_DOMAINS.includes(
+          `${protocol}//${hostname}`
+        );
```

Comment:
```suggestion
        const { origin } = new URL(url);

        const isDomainValid = WHITELISTED_DOMAINS.includes(origin);
```
Have you tried with `origin` parameter? Could it work?

Reply:
yesss done! Good suggestion!